### DISCUSSION TITLE
Wrong Bigint value in API request using pagination

### DISCUSSION THREAD

**Discussion by tienrocker** [laravel/framework#33316] [General]
- Laravel Version: 7.16.1
- PHP Version: 7.2.31

### Description:

I use pagination to return in api response and it show wrong value of bigint fields. Example below:

```json
{
   "current_page":200,
   "data":[
      {
         "id":104253013262866000,
         "league_id":101097443346691680,
         "tournament_id":104253013242629780,
         "team1_id":104251016988118460,
         "team2_id":103535282148790980
      }
   ]
}
```

In model file, i cast field to string then it work

```json
{
   "current_page":200,
   "data":[
      {
         "id":"104253013262865997",
         "league_id":"101097443346691685",
         "tournament_id":"104253013242629769",
         "team1_id":"104251016988118464",
         "team2_id":"103535282148790975"
      }
   ]
}
```

My dump value still show right int value, example `dd($items[0]->toArray())`

### Steps To Reproduce:

```php
public function index()
{
    $items = TABLE_NAME::query()->paginate($limit);
    return $items;
}
```

**Comment by Lunaversitay** [comment]
- Laravel `7.9.2`
- PHP `7.3.2`
- PHP_INT_SIZE `8`

Pretty convenient, I'm actually having the same issue with using bigint on the resources.
In my case it keeps incrementing the bigints I give it by 1

This is the actual bigint I'm trying to use:
`76561198013448859`

This is what I get in return:
`76561198013448860`

However this doesn't just occur on pagination-ed collections, it also occurs on literally any resource.

At first me and a friend thought maybe PHP couldn't process the bigint but dd/die debugging it displays the correct number.

Edit: After posting this a friend let me know this will happen when you serialize to json - best solution so far is to just cast to a string I suppose. He sent this as an explanation: https://tools.ietf.org/html/rfc7159#section-6
